### PR TITLE
Fix issue 1703: symmetric equality

### PR DESCRIPTION
**Pull request by mkooiman**
This PR fixes:
https://github.com/JasperFx/marten/issues/1703

Also includes several tests to test for this issue

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/JasperFx/marten?pullRequest=1706) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/JasperFx/marten?pullRequest=1706) before we can accept your contribution.<br/><br/>:x: mkooiman [sign now](https://cla.dotnetfoundation.org/JasperFx/marten?pullRequest=1706)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/JasperFx/marten?pullRequest=1706) it.</sub>

**Comment by mysticmind** [comment]
@mkooiman CI build are failing, please have a look at [build log](https://github.com/JasperFx/marten/pull/1706/checks?check_run_id=1705106637)

File: src/Marten/Linq/Parsing/Methods/StringComparisonParser.cs [hunk 1]
```diff
@@ -28,9 +28,20 @@ public ISqlFragment Parse(IFieldMapping mapping, ISerializer serializer, MethodC
         {
             var locator = GetLocator(mapping, expression);
 
-            var value = expression.Arguments.OfType<ConstantExpression>().FirstOrDefault();
+            ConstantExpression value;
+            if (expression.Object.NodeType == ExpressionType.Constant)
```
Comment (mysticmind) [inline src/Marten/Linq/Parsing/Methods/StringComparisonParser.cs:32]: You will need to add a null check `expression.Object?.NodeType == ExpressionType.Constant` to fix the failing tests.

Comment (mkooiman) [reply src/Marten/Linq/Parsing/Methods/StringComparisonParser.cs]: Awesome, thanks! I was going to look in to it but didn't have much time to spare.

Comment (mysticmind) [reply src/Marten/Linq/Parsing/Methods/StringComparisonParser.cs]: No worries, thanks very much for the PR and the tests 🏆

**Review by mysticmind — APPROVED** [review]
Looks good to me 👍

**Comment by mysticmind** [comment]
@mkooiman Quick note, I will take care of adding the same fix to 3.x as well.

**Review by oskardudycz — APPROVED** [review]

**Review by jeremydmiller — APPROVED** [review]
LGTM. We *could* think about centralizing the symmetric equality to more places than the string comparison here though. Later time.

**Pull request merged**